src/alt/index.js

```diff
@@ -138,10 +145,28 @@ class Alt {
       assign(actions, ActionsClass)
     }
 
+    this.actions[key] = this.actions[key] || {}
+
     return Object.keys(actions).reduce((obj, action) => {
-      obj[action] = this.createAction(`${key}#${action}`, actions[action], obj)
+      if (typeof actions[action] !== 'function') {
+        return obj
+      }
+
+      // create the action
+      const name = `${key}#${action}`
+      obj[action] = this.createAction(name, actions[action], obj)
+
+      // remove the action referenced in single
```

Comment:
This feels a bit round about to me.  The fact that creating the action always puts it on the `single` property seems inflexible. Maybe we can pass in the key on actions that we want to attach it to?